internal/controller/nodepool_controller_test.go

```diff
@@ -147,12 +148,17 @@ func TestReconcileNodePool(t *testing.T) {
 
 	// delete the NodePool object and check if the nodes are deleted
 	err = fakeClient.Get(ctx, types.NamespacedName{Name: "single-nodepool"}, nodePool)
+	if err != nil {
+		log.Println("failed to get NodePool object")
+	}
 	err = fakeClient.Delete(ctx, nodePool)
+	if err == nil {
```

Comment:
Should be "if err != nil"

Reply:
Nope , it should be like that , if the error equal nil it means the nodePool still exist and it's not what i'm expecting

Reply:
what is the output if it succeed to delete?